bokehjs/src/lib/core/property_mixins.ts

```diff
@@ -18,6 +18,15 @@ export type Fill = {
   fill_alpha: p.Property<number>
 }
 
+export type Hatch = {
+  hatch_color: p.Property<Color | null>
+  hatch_alpha: p.Property<number>
+  hatch_scale: p.Property<number>
+  hatch_pattern: p.Property<string>
```

Comment:
Should be `HatchPatternType` instead of `string`.

Reply:
let it string for now, given requirements for custom textures. Open to other ideas to tighten things up.

Reply:
I also used `p.Instance` in `hatch_extra` it seems like this module should not depend on models imports

Reply:
At least for now, I would add `export type HatchPattern = HatchPatternType | string`. This doesn't increase type safety, but at least indicates more clearly the intent. I would also mark `hatch_extra` and texture support in `hatch_pattern` as experimental (similarly to `width_policy` or `height_policy` in `LayoutDOM`), so that there are no issues with API breakage if a refined API pops up in near future.